pkgs/dart_model/lib/src/json_changes.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'json.dart';
+
+/// The changes between two [Json] instances.
+extension type JsonChanges.fromJson(Map<String, Object?> node) {
```

Comment:
Oof, storing this as an object with keys is physically painful to me 😬

Reply:
;) just keep saying "it's an efficient binary representation underneath, with deduped strings" ... I do think it will be.

If you think _this_ is bad, wait until you see "Properties", a Map from "abstract", "static", etc to bools. It's going to take up a lot of space in the human readable format. I'm really torn between correctness of schema and wanting to do anything at all to make it smaller ... it would also probably get more readable! But we can make the verbose form efficient too. Let's bikeshed that one later ;)